server/events.go

```diff
@@ -833,46 +833,87 @@ func (s *Server) leafNodeConnected(sub *subscription, _ *client, subject, reply
 }
 
 // Common filter options for system requests STATSZ VARZ SUBSZ CONNZ ROUTEZ GATEWAYZ LEAFZ
-type ZFilterOptions struct {
-	Name    string `json:"name"`
-	Cluster string `json:"cluster"`
-	Host    string `json:"host"`
+// These options are
+type EventFilterOptions struct {
+	Name    string `json:"server_name,omitempty"` // filter by server name
+	Cluster string `json:"cluster,omitempty"`     // filter by cluster name
+	Host    string `json:"host,omitempty"`        // filter by host name
+}
+
+// StatszEventOptions are options passed to Statsz
+type StatszEventOptions struct {
+	// No actual options yet
+
+	EventFilterOptions
+}
+
+// In the context of system events, ConnzEventOptions are options passed to Connz
+type ConnzEventOptions struct {
+	ConnzOptions
+	EventFilterOptions
+}
+
+// In the context of system events, RoutezEventOptions are options passed to Routez
+type RoutezEventOptions struct {
+	RoutezOptions
+	EventFilterOptions
+}
+
+// In the context of system events, SubzEventOptions are options passed to Subz
+type SubszEventOptions struct {
+	SubszOptions
+	EventFilterOptions
+}
+
+// In the context of system events, VarzEventOptions are options passed to Varz
+type VarzEventOptions struct {
+	VarzOptions
+	EventFilterOptions
+}
+
+// In the context of system events, GatewayzEventOptions are options passed to Gatewayz
+type GatewayzEventOptions struct {
+	GatewayzOptions
+	EventFilterOptions
+}
+
+// In the context of system events, LeafzEventOptions are options passed to Leafz
+type LeafzEventOptions struct {
+	LeafzOptions
+	EventFilterOptions
 }
 
 // returns true if the request does NOT apply to this server and can be ignored.
 // DO NOT hold the server lock when
-func (s *Server) filterRequest(msg []byte) (bool, error) {
-	if len(msg) == 0 {
-		return false, nil
-	}
-	var fOpts ZFilterOptions
-	err := json.Unmarshal(msg, &fOpts)
-	if err != nil {
-		return false, err
+func (s *Server) filterRequest(fOpts *EventFilterOptions) bool {
+	if fOpts == nil {
+		return false
 	}
 	if fOpts.Name != "" && !strings.Contains(s.info.Name, fOpts.Name) {
-		return true, nil
+		return true
 	}
 	if fOpts.Host != "" && !strings.Contains(s.info.Host, fOpts.Host) {
-		return true, nil
+		return true
 	}
 	if fOpts.Cluster != "" {
 		s.mu.Lock()
 		cluster := s.info.Cluster
 		s.mu.Unlock()
 		if !strings.Contains(cluster, fOpts.Cluster) {
-			return true, nil
+			return true
 		}
 	}
-	return false, nil
+	return false
 }
 
-// statszReq is a request for us to respond with current statz.
+// statszReq is a request for us to respond with current statsz.
 func (s *Server) statszReq(sub *subscription, _ *client, subject, reply string, msg []byte) {
 	if !s.EventsEnabled() || reply == _EMPTY_ {
 		return
 	}
-	if ignore, err := s.filterRequest(msg); err != nil {
+	opts := StatszEventOptions{}
+	if len(msg) == 0 {
```

Comment:
Valid but unusual in server source code. Although your approach reduces indentation, I think this would be preferred:
```
if len(msg) != 0 {
 ... unmarshal and filter code would go inside this if
}
```

Reply:
changed